### PR TITLE
feat: Add line highlighting for syntax errors

### LINKED ISSUE #6979: Highlight the syntax error in the cell
### Description

When a cell doesn't parse, and it shows an error above the cell (`This cell wasn't run because it has errors ... line ...`), it would be convenient to also highlight the line mentioned in the error.

### Suggested solution

Highlight syntax errors line with the same highlight as exception line highlights.

### Are you willing to submit a PR?

- [x] Yes

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SamarJyoti496**
## 📝 Summary

Syntax Errors now show red line highlighting in the editor matching the existing behavior for runtime errors.

## 🔍 Description of Changes
**Problem:** Syntax errors displayed error messages but didn't highlight the problematic line in red, unlike runtime errors.

**Solution:** Extract line numbers from Python's `SyntaxError.lineno` attribute and feed them into the existing traceback.

## 📋 Checklist

- [x] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [x] I have run the code and verified that it works as expected.

---

> [!NOTE]
> Extracts `lineno` for Python syntax/import-star errors in the runtime, exposes it via API, and updates frontend traceback logic to highlight exact error lines, with comprehensive tests.
> 
> - **Backend**:
>   - Add optional `lineno` to `MarimoSyntaxError` and `ImportStarError`; capture from `SyntaxError.lineno` in runtime.
> - **API**:
>   - Extend OpenAPI schema and generated TS (`packages/openapi`) to include `lineno` on `MarimoSyntaxError` and `ImportStarError`.
> - **Frontend**:
>   - Refactor `createTracebackInfoAtom` to use `cellRuntimeAtom` and parse `syntax`/`import-star` errors with `lineno` into `TracebackInfo` (skip when queued/running).
>   - Enhance mocks to allow per-cell `cellRuntime` overrides.
>   - Export `createTracebackInfoAtom` for testing.
> - **Tests**:
>   - Add/expand tests validating `lineno` propagation (including 0, null, multiline) and new traceback behavior; update serialization expectations to include `lineno`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9da2233276bdc3d0781d7372a0d6306317986ecb. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: marimo/_runtime/runtime.py [hunk 1]
```diff
@@ -875,10 +875,14 @@ def _try_compiling_cell(
                 syntax_error[0] = syntax_error[0][
                     syntax_error[0].find("line") :
                 ]
+
+                lineno = getattr(e, "lineno", None)
                 if isinstance(e, ImportStarError):
                     error = MarimoImportStarError(msg=str(e))
                 else:
-                    error = MarimoSyntaxError(msg="\n".join(syntax_error))
+                    error = MarimoSyntaxError(
```
Comment (mscolnick) [inline marimo/_runtime/runtime.py:884]: if there are any tests that check the a `MarimoSyntaxError` was created, can you also check that `lineno` was also added

File: frontend/src/core/cells/cells.ts [hunk 2]
```diff
@@ -1689,40 +1689,51 @@ export function createUntouchedCellAtom(cellId: CellId): Atom<boolean> {
...
+    const tracebackInfo: TracebackInfo[] = [];
+
+    // Runtime errors (ZeroDivisionError, etc.)
     const outputs = data.consoleOutputs;
-    // console.warn(notebook);
-    if (!outputs || outputs.length === 0) {
-      return undefined;
+    if (outputs && outputs.length > 0) {
+      const firstTraceback = outputs.find(
+        (output: any) => output.mimetype === "application/vnd.marimo+traceback",
```
Comment (mscolnick) [inline frontend/src/core/cells/cells.ts]:
```suggestion
        (output) => output.mimetype === "application/vnd.marimo+traceback",
```

File: frontend/src/core/cells/cells.ts [hunk 3]
```diff
@@ -1689,40 +1689,51 @@ export function createUntouchedCellAtom(cellId: CellId): Atom<boolean> {
 export function createTracebackInfoAtom(
   cellId: CellId,
 ): Atom<TracebackInfo[] | undefined> {
-  // We create an intermediate atom that just computes the string
-  // so it prevents downstream recomputations.
-  const tracebackStringAtom = atom<string | undefined>((get) => {
+  // Single flat atom - proper reactivity
+  return atom((get: any) => {
```
Comment (mscolnick) [inline frontend/src/core/cells/cells.ts]: this now runs on every notebook change. can we instead at least do keep the intermediate atom that computes `cellDataAtom` (just returning `const data = notebook.cellRuntime[cellId]`)

**Comment by mscolnick** [comment]
@SamarJyoti496 you will also need to run `make fe-codegen`. this might create some newlines in areas that are unrealted. you can revert parts of that change as we look into why it does that

**Comment by SamarJyoti496** [comment]
> @SamarJyoti496 you will also need to run `make fe-codegen`. this might create some newlines in areas that are unrealted. you can revert parts of that change as we look into why it does that

I have updated that one. Some backend test cases failed, but I'm not sure whether they're related to my changes. I think they're not as they had run successfully before.

**Comment by mscolnick** [comment]
@SamarJyoti496 im going to push some changes for some tests and a bit of cleanup
